.github/workflows/prepare-matrix.yml

```
@@ -0,0 +1,54 @@
+name: Prepare Matrix
+
+on:
+  workflow_dispatch:
+    inputs:
+      matrix_filename:
+        description: 'Matrix of jobs and their respective environments'
+        required: false
+        default: 'quic_matrix.json'
+        type: string
+      workflowId:
+        description: 'The ID of the caller workflow'
+        required: false
+        default: 'manual'
+        type: string
+
+  workflow_call:
+    inputs:
+      matrix_filename:
+        description: "Matrix of jobs and their respective environments"
+        required: false
+        default: 'quic_matrix.json'
+        type: string
+      workflowId:
+        description: "The ID of the caller workflow"
+        required: true
+        type: string
+    outputs:
+      full-matrix:
+        description: "The full list of jobs to be run in secnetperf"
+        value: ${{ jobs.prepare-matrix.outputs.full-matrix }}
+      azure-matrix:
+        description: "The list of jobs to be run on Azure VMs"
+        value: ${{ jobs.prepare-matrix.outputs.azure-matrix }}
+permissions: write-all
```

nit: add extra newline
```suggestion

permissions: write-all
```